parallel-install/pkg/deployment/overrideinterceptors.go

```diff
@@ -25,20 +31,82 @@ type OverrideInterceptor interface {
 	Undefined(overrides map[string]interface{}, key string) error
 }
 
+func NewDomainNameOverrideInterceptor(kubeClient kubernetes.Interface, retryOptions []retry.Option, log logger.Interface) *DomainNameOverrideInterceptor {
+	return &DomainNameOverrideInterceptor{
+		kubeClient:   kubeClient,
+		retryOptions: retryOptions,
+		log:          log,
+	}
+}
+
 //DomainNameOverrideInterceptor resolves the domain name for the cluster
 type DomainNameOverrideInterceptor struct {
+	kubeClient   kubernetes.Interface
+	retryOptions []retry.Option
+	log          logger.Interface
 }
 
 func (i *DomainNameOverrideInterceptor) String(value interface{}, key string) string {
 	return fmt.Sprintf("%v", value)
 }
 
 func (i *DomainNameOverrideInterceptor) Intercept(value interface{}, key string) (interface{}, error) {
+	//on gardener domain provided by user should be ignored
+	domainName, err := i.getGardenerDomain()
+	if err != nil {
+		return nil, err
+	}
+
+	if domainName != "" {
+		return domainName, nil
+	}
+
 	return value, nil
 }
 
 func (i *DomainNameOverrideInterceptor) Undefined(overrides map[string]interface{}, key string) error {
-	return NewFallbackOverrideInterceptor(localKymaDevDomain).Undefined(overrides, key)
+	domain, err := i.discoverDomain()
+	if err != nil {
+		return err
+	}
+
+	return NewFallbackOverrideInterceptor(domain).Undefined(overrides, key)
+}
+
+func (i *DomainNameOverrideInterceptor) discoverDomain() (string, error) {
+	domainName, err := i.getGardenerDomain()
+	if err != nil {
+		return "", err
+	}
+
+	if domainName != "" {
+		return domainName, nil
+	}
+	return localKymaDevDomain, nil
+}
+
+func (i *DomainNameOverrideInterceptor) getGardenerDomain() (string, error) {
+	domainName := ""
+	err := retry.Do(func() error {
+		configMap, err := i.kubeClient.CoreV1().ConfigMaps("kube-system").Get(context.TODO(), "shoot-info", metav1.GetOptions{})
+
+		if err != nil {
+			if apierr.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+
+		domainName = configMap.Data["domain"]
```

Comment:
Is there a possibility of `nil` if this property does not exist?

Reply:
i don't think so but we can double check and throw an error

Reply:
There is no possibility of `nil`, only empty string ("") since the zero value of a string is "".

Alternatively you can also check like this (more safe):
```suggestion
		domainName, exists := configMap.Data["domain"]
		if !exists {
		  // return an error
		}
```